system/jlib/jstring.cpp

```diff
@@ -1056,13 +1056,27 @@ StringBuffer &replaceStringNoCase(StringBuffer & result, size_t lenSource, const
 // this method will replace all occurrences of "oldStr" with "newStr"
 StringBuffer & StringBuffer::replaceString(const char* oldStr, const char* newStr)
 {
-    if (curLen)
+    if (curLen && oldStr)
     {
-        StringBuffer temp;
-        size_t oldlen = oldStr ? strlen(oldStr) : 0;
+        size_t oldlen = strlen(oldStr);
+        if (oldlen > curLen)
+            return *this;
+
         size_t newlen = newStr ? strlen(newStr) : 0;
-        ::replaceString(temp, curLen, buffer, oldlen, oldStr, newlen, newStr);
-        swapWith(temp);
+        if (oldlen == curLen)
+        {
+            if (memcmp(buffer, oldStr, oldlen) == 0)
+            {
+                memcpy(buffer, newStr, newlen);
```

Comment:
If I was reviewing this code (I am not because of general review comments)  - this has a serious memory corruption if newlen > oldlen

Reply:
Fixed. I added ensureCapacity(newlen).